avocado/core/status/repo.py

```diff
@@ -53,10 +59,25 @@ def get_latest_task_data(self, task_id):
             return None
         return task_data[-1]
 
+    def _update_status(self, message):
+        """Update the latest status of atask (by time, not by message)."""
+        task_id = message.get('id')
+        status = message.get('status')
+        time = message.get('time')
+        if not all((task_id, status, time)):
+            return
+        if task_id not in self._status:
+            self._status[task_id] = (status, time)
+        else:
+            current_time = self._status[task_id][1]
+            if time > current_time:
+                self._status[task_id] = (status, time)
```

Comment:
To avoid if else here, one suggestions is to have inside a try.. except:

```python
try:
    if time > self._status[task_id][1]:
        self._status[task_id] = (status, time)
except KeyError:
    self._status[task_id] = (status, time)
```

Reply:
It's more compact indeed... but I must say I find the readability a bit worse (my bias, of course).  Thanks for the suggestion, though.